### PR TITLE
Authority refactor

### PR DESCRIPTION
**Pull request by sameerag**
First of the authority related changes.

This PR will be followed by:

- B2C authority: removing 'validateAuthority' check for B2C authority
- instance_aware
- Unit tests

Note: This PR replaces #951 for a cleaner merge with dev

### CODE REVIEW

File: lib/msal-core/src/authority/B2cAuthority.ts [hunk 1]
```diff
@@ -32,19 +32,11 @@ export class B2cAuthority extends AadAuthority {
...
-
-        if (!this.IsValidationEnabled) {
-            return resultPromise;
-        }
-
-        if (this.IsInTrustedHostList(this.CanonicalAuthorityUrlComponents.HostNameAndPort)) {
-            return resultPromise;
+    public async GetOpenIdConfigurationEndpointAsync(): Promise<string> {
+        if (!this.IsValidationEnabled || this.IsInTrustedHostList(this.CanonicalAuthorityUrlComponents.HostNameAndPort)) {
+            return this.DefaultOpenIdConfigurationEndpoint;
```
Comment (DarylThayil) [inline lib/msal-core/src/authority/B2cAuthority.ts]: this need to return a promise doesn't it ?

return new Promise.resolve(this.defaultEndpoint)

Comment (jasonnutter) [reply lib/msal-core/src/authority/B2cAuthority.ts]: Its an `async` function so it does inherently.

Comment (sameerag) [reply lib/msal-core/src/authority/B2cAuthority.ts]: Since it is an `async` function, all return statements are wrapped as promises from my understanding.

Comment (DarylThayil) [reply lib/msal-core/src/authority/B2cAuthority.ts]: ahh okay makes sense

File: lib/msal-core/src/UserAgentApplication.ts [hunk 2]
```diff
@@ -1772,6 +1763,23 @@ export class UserAgentApplication {
...
+
+    /**
+     * Set Authority when saving Token from the hash
+     * @param state
+     * @param inCookie
+     * @param cacheStorage
+     * @param idTokenObj
+     * @param response
+     */
+    private populateAuthtority(state: string, inCookie: boolean, cacheStorage: Storage, idTokenObj: IdToken): string {
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: Typo here

Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: Should this be in Authority.ts?

Comment (sameerag) [reply lib/msal-core/src/UserAgentApplication.ts]: Fixed. Please check.

Comment (sameerag) [reply lib/msal-core/src/UserAgentApplication.ts]: nah! We are setting it in UAA for UAA reference. There are many small references to different response/cache elements in UAA we want to refactor as a part of saveTokenFromHash(). If Authority.ts feels right then, we can do it.

**Comment by DarylThayil** [comment]
@sameerag  can this be merged

**Comment by sameerag** [comment]
@DarylThayil Merged now

**Pull request merged**